theano/sandbox/cuda/opt.py

```diff
@@ -300,6 +300,25 @@ def local_gpu_elemwise_1(node):
 
 
 @register_opt()
+@local_optimizer([tensor.Split, gpu_from_host])
+def local_gpu_split(node):
+    if isinstance(node.op, tensor.Split):
+        input = node.inputs[0]
+        if input.owner and isinstance(input.owner.op, HostFromGpu):
+            new_op = GpuSplit(node.op.len_splits)
+            split_res = new_op(gpu_from_host(input), *node.inputs[1:])
+            return [host_from_gpu(o) for o in split_res]
+    if isinstance(node.op, GpuFromHost):
```

Comment:
As the Split op have many outputs, handing this case is more complicated. I would suggest to just remove this if completly, we do not need it frequently. In the new gpu back-end it won't be needed, so doing the compliated case isn't very useful.